src/unleash-proxy.ts

```diff
@@ -293,12 +293,20 @@ export default class UnleashProxy {
         } else {
             const { context = {}, toggles: toggleNames = [] } = req.body;
 
-            const toggles = this.client.getDefinedToggles(
-                toggleNames,
-                context as Context,
-            );
-
-            res.send({ toggles });
+            if (toggleNames.length > 0) {
+                const toggles = this.client.getDefinedToggles(
+                    toggleNames,
+                    context as Context,
+                );
+                res.set('Cache-control', 'public, max-age=2');
+                res.send({ toggles });
+            } else {
+                context.remoteAddress = context.remoteAddress || req.ip;
+                const actualContext = createContext(context);
+                const toggles = this.client.getEnabledToggles(actualContext);
+                res.set('Cache-control', 'public, max-age=2');
+                res.send({ toggles });
+            }
```

Comment:
It's a bit confusing that we keep the same functionality if you send toggleNames, but anything else would be a breaking change. Would it make sense as it's own endpoint, or will that break other paths?

There might be an OpenAPI spec we should update aswell as a part of this change?

Reply:
> It's a bit confusing that we keep the same functionality if you send toggleNames, but anything else would be a breaking change. Would it make sense as it's own endpoint, or will that break other paths?

I agree, but at the same time it feels a bit strange to have yet another path, and it will require more changes to the SDKs


> There might be an OpenAPI spec we should update aswell as a part of this change?
I think it was already optional to provide the feature toggle names, I will check.